File: Greetings App/app/src/main/java/com/example/greetingsapp/MainActivity.java
package com.example.greetingsapp;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;
import android.widget.Toast;

public class MainActivity extends AppCompatActivity {

    EditText editText;
    Button myBtn;
    TextView title;


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        myBtn = findViewById(R.id.myBtn);
        title = findViewById(R.id.title);
        editText = findViewById(R.id.editText);

        myBtn.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                String input = editText.getText().toString();
                Toast.makeText(MainActivity.this, "Welcome " + input, Toast.LENGTH_SHORT).show();
            }
        });

    }
}

File: DialogFragment/app/src/main/java/com/tiramisu/dialogfragment/MainActivity.kt
package com.tiramisu.dialogfragment

import android.os.Bundle
import android.widget.Button
import android.widget.TextView
import androidx.activity.enableEdgeToEdge
import androidx.appcompat.app.AppCompatActivity
import androidx.core.view.ViewCompat
import androidx.core.view.WindowInsetsCompat
import androidx.fragment.app.FragmentManager

class MainActivity : AppCompatActivity() {

    lateinit var show : Button
    lateinit var name : TextView
    lateinit var age : TextView

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        enableEdgeToEdge()
        setContentView(R.layout.activity_main)
        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main)) { v, insets ->
            val systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars())
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom)
            insets
        }

        show = findViewById(R.id.button)
        name = findViewById(R.id.tvName)
        age = findViewById(R.id.tvAge)

        show.setOnClickListener {
            val fragmentManager : FragmentManager= supportFragmentManager
            val myDialogFragment = MyDialogFragment()
            myDialogFragment.show(fragmentManager, "MyDialogFragment")
            myDialogFragment.isCancelable = false// cancelled when tapped outside the dialog box
        }
    }

    fun getUserData(userName : String , userAge : Int){
        name.text = userName
        age.text = userAge.toString()
    }
}